### PR TITLE
Feature/doc 181

### PR DESCRIPTION
**Pull request by shikhanansi**
Hi Millie,
Please add your review feedback to this PR.
Regards,
Shikha

### CODE REVIEW

File: migration-from-4x.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+Our team at Tiny has worked on creating a more configurable, more modern and less cumbersome editor while retaining the familiarity of the user interface from the older versions.
+
+This chapter describes the migration process and workarounds if you are using an older version of TinyMCE. It includes tasks that you must perform before the migration can begin, and different workaround procedures for deprecated, deleted, and updated features.
+
+## Editor-Core
+
+### Initialization
+
+The initialization process of TinyMCE 5.0 is the same as TinyMCE 4.x. The bootstrap process and initialization events all remain the same.  The main differences are in the `init` configuration, specifically the configuration items for UI components. It still retains a familiar JSON structure. However, many of the configuration options have been simplified.
```
Comment (blackcathikari) [inline migration-from-4x.md]: - "initialisation process" should really be quantified - what does this refer to? Adding the TinyMCE 5.0 script to the page? Setting up a basic config? 
- What do we mean by "bootstrap process and initialisation events"? Are these terms from somewhere else in the docs, in which case can we link to it for clarity?

File: migration-from-4x.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+### Settings
+
+In TinyMCE 5.0, some configurations have been removed because they are no longer necessary or an improved solution has been introduced.
```
Comment (blackcathikari) [inline migration-from-4x.md]: Is this blurb still needed? If so, we should say "...some configurations have been changed or removed because..." and also mention new settings, if we added any.

File: migration-from-4x.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+## Themes
+
+In TinyMCE 5.0, some themes have been removed and are now combined in a new single responsive theme called **Silver**.
+
+| **Removed Theme** | **Replaced by** |
+| ----------------- | --------------  |
+| [Modern](https://www.tiny.cloud/docs/themes/modern/) | Silver |
+| [Modern inline](https://www.tiny.cloud/docs/general-configuration-guide/use-tinymce-inline/) | Silver Inline |
+| [Inlite(Distraction-free Editor)](https://www.tiny.cloud/docs/themes/inlite/) | Silver (distraction free configuration) |
+| [Mobile](https://www.tiny.cloud/docs/themes/mobile/) | Silver (responsive to screen size)  |
```
Comment (blackcathikari) [inline migration-from-4x.md]: @EkimChau Please confirm this table is still correct, re: mobile.

File: migration-from-4x.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+#### Modern
+
+The Modern theme is no longer supported in TinyMCE 5.0.  The modern themes Ui library `tinymce.ui.*` has also been deleted. This change may impact integrations depending upon the level of customization.
+
+For changes required, refer to the following table of examples:
+
+| Customization Level | Description | Impact |
+| ------------------- | ----------- | ------ |
+| Minor | Some custom buttons | no Ui fixes required, update `addButton` configuration to TinyMCE 5.0 format |
```
Comment (blackcathikari) [inline migration-from-4x.md]: Capitalise "no" and add a link to custom components docs?

File: migration-from-4x.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+| Moderate | A dialog created using `editor.windowManager.open` configuration objects | Convert TinyMCE 4.x config to TinyMCE 5.0 config |
```
Comment (blackcathikari) [inline migration-from-4x.md]: Improve Impact comment and link to docs for making custom dialogs?

File: migration-from-4x.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+| Major | Completely custom dialogs and extended use of the Modern UI framework | Not all API use cases are covered by our new TinyMCE 5.0 components. However, we will strive to create supported workarounds or if there are sufficient requests, add a new component to resolve the use case. |
```
Comment (blackcathikari) [inline migration-from-4x.md]: Can we link to the Github page for TinyMCE 5.0 tickets?

File: migration-from-4x.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+
+> Note: Please provide feedback on your use case and your Tinymce 4.x configuration containing only the UI component that you wish to be supported or need a workaround.
+
+> Note: The Silver theme in TinyMCE 5.0 contains a set of configurable UI components that could be used to replace the current customizations (modern, inline, and inlite theme). Silver is enabled by default. If you do not specify a theme, it will default to Silver.
```
Comment (blackcathikari) [inline migration-from-4x.md]: The first sentence of this note probably needs clarification. To me, it seems unrelated to the rest of the paragraph at the moment?

File: migration-from-4x.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+
+| **New Method** | **Description** |
+| ----------- | -------------- |
+| editor.ui.registry.addContextMenu() | Adds a custom context menu. |
+
+For more information on Context Menus, see the [docs]({{site.baseurl}}/components/contextmenu).
+
+### Custom Context Forms
+
+ContextForms are a generalisation of the `Insert Link` form that existed in the original `inlite` plugin from [TinyMCE 4.x]((https://www.tiny.cloud/docs/themes/inlite/#quicklink)).
```
Comment (blackcathikari) [inline migration-from-4x.md]: We also have a page for custom context forms, we should link to that too?

File: migration-from-4x.md [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+#### Changes between TinyMCE 4.x and TinyMCE 5.0:
+
+* Buttons go before and after the input in TinyMCE 4.x.
+* The `Ctrl+K` shortcut does nothing until the context toolbar is visible in TinyMCE 4.x.
+* In TinyMCE 5.0, the pop animates to its new width.
+* In TinyMCE 4.x., it is a URL input, so you get a popup and a browse button.
+
+### Custom Context Menus
+
+The default context menu is no longer a plugin, it is part of the core and always enabled. Where TinyMCE 4.x only supported adding registered menu items, the new context menu also allows plugins to register "sections" of the context menu. These sections are dynamic and may show or hide depending on the cursor position when the context menu is opened. For example, the default context menu config is now `'link image editimage table spelling'` which are all plugin section references, not menu items.
```
Comment (blackcathikari) [inline migration-from-4x.md]: I think the first sentence needs to be expanded, or another sentence needs to be added. It doesn't flow to the next sentence.

File: migration-from-4x.md [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+
+Docs coming soon!
+
+### Custom Context Toolbars
+
+The Context Toolbar takes its buttons the registry of toolbar buttons added using `addButton`, `addToggleButton`, `addSplitButton` or `addMenuButton`. The method for creating custom context toolbars has also been moved from `editor.addContextToolbar()` to `editor.ui.registry.addContextToolbar()`.
+
+For more information on Context Toolbars, see [docs]({{site.baseurl}}/components/contexttoolbar).
+
+#### Changes between TinyMCE 4.x and TinyMCE 5.0:
```
Comment (blackcathikari) [inline migration-from-4x.md]: This section mostly only lists TinyMCE 4.x behaviours, it doesn't say what's changed. @shikhanansi @EkimChau We need to chase up what the actual difference is. 

Also, in the third point what does "pop" refer to?

File: migration-from-4x.md [hunk 11]
```diff
@@ -0,0 +1,328 @@
...
+| ----------- | -------------- |
+| editor.ui.registry.addContextMenu() | Adds a custom context menu. |
+
+For more information on Context Menus, see the [docs]({{site.baseurl}}/components/contextmenu).
+
+### Custom Context Forms
+
+ContextForms are a generalisation of the `Insert Link` form that existed in the original `inlite` plugin from [TinyMCE 4.x]((https://www.tiny.cloud/docs/themes/inlite/#quicklink)).
+
+### Toolbar Menus
```
Comment (blackcathikari) [inline migration-from-4x.md]: This section should probably be deleted or combined into the Toolbar Buttons section above, as I believe it refers to custom toolbar menu buttons? I'm not sure what the first point in this list is referring to though. @EkimChau do you know?

File: migration-from-4x.md [hunk 12]
```diff
@@ -0,0 +1,328 @@
...
+### Changed Features
+
+These features have either changed or have been deleted in TinyMCE 5.0.
+Each release of TinyMCE adds new features and functionality. We also occasionally remove features and functionality, usually because we’ve added a better option. Here are the details about the features and functionalities that we removed in TinyMCE 5.0.
+
+#### Moved Plugins
+
+| **Plugin Name** | **Description** |
+| --------------- |  -------------- |
+| [ContextMenu](https://www.tiny.cloud/docs/plugins/contextmenu/) | New API. See [docs]({{site.baseurl}}/components/contextmenu/). |
```
Comment (blackcathikari) [inline migration-from-4x.md]: I think we need to add colorpicker to this somewhere? @EkimChau ?

File: migration-from-4x.md [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+
+* Styles text field has been removed from the advanced table of the dialogs. This simplifies the dialogs for users and gives the editor stricter control over the table styles which means we are better able to ensure the styles are correct.
+* Improved how styles are set and retrieved from tables, rows, and cells, so this should be more reliable now.
+* Shifted to using CSS more for styling, and therefore removed a few legacy data attributes that were set on tables/rows/cells which are no longer the good practice to use. This makes the output HTML cleaner and more modern.
+* When opening a properties dialog with a single table/row/cell selected, the dialog will autofill with the relevant existing values. If you select multiple rows or cells and open the relevant properties dialog, TinyMCE 4.x will leave all the dialog fields blank. In TinyMCE 5.0, any fields which have the same values for all the selected rows or cells will autofill, and the fields which have no existing value or have differing values will be empty.
+* `Border` input field in the `tableprops` dialog is now called `Border width`, for better clarity.
+
+
+## Mobile Support
```
Comment (blackcathikari) [inline migration-from-4x.md]: @EkimChau Please confirm this section is still correct.

File: migration-from-4x.md [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+
+The initialization process of TinyMCE 5.0 is the same as TinyMCE 4.x. The bootstrap process and initialization events all remain the same.  The main differences are in the `init` configuration, specifically the configuration items for UI components. It still retains a familiar JSON structure. However, many of the configuration options have been simplified.
+
+### Settings
+
+In TinyMCE 5.0, some configurations have been removed because they are no longer necessary or an improved solution has been introduced.  All inline style configurations have been removed in TinyMCE 5.0 in favour of modern CSS.
+
+#### Changed Settings
+
+| **Old Settings** | **Description** | **Alternative** |
```
Comment (blackcathikari) [inline migration-from-4x.md]: Could we add a blurb to say where these settings came from in TinyMCE 4.x? It's a bit confusing at the moment, to just have a list of what looks like CSS styles...

Comment (blackcathikari) [reply migration-from-4x.md]: Also, should these go under Removed Settings rather than Changed Settings?

Comment (EkimChau) [reply migration-from-4x.md]: yes will re-edit the text for clarity

yeah thats true, they are a removed setting. We should move this to a removed settings

File: migration-from-4x.md (hunk 11 above)
Comment (EkimChau) [reply migration-from-4x.md]: yeah let me reword

File: migration-from-4x.md (hunk 10 above)
Comment (EkimChau) [reply migration-from-4x.md]: We should rewrite this summary at the end, i've yet to review all the other change docs

File: migration-from-4x.md [hunk 15]
```diff
@@ -0,0 +1,325 @@
...
+
+The initialization process of TinyMCE 5.0 is the same as TinyMCE 4.x. The bootstrap process and initialization events all remain the same.  The main differences are in the `init` configuration, specifically the configuration items for UI components. It still retains a familiar JSON structure. However, many of the configuration options have been simplified.
+
+### Settings
+
+In TinyMCE 5.0, some configurations have been removed because they are no longer necessary or an improved solution has been introduced.  All inline style configurations have been removed in TinyMCE 5.0 in favour of modern CSS.  This affects all TinyMCE 4 Ui component configurations.
+
+#### Removed Settings
+
+* **affects all TinyMCE 4 ui component configurations**
```
Comment (blackcathikari) [inline migration-from-4x.md]: Capitalise. Also, should this be a heading, like "Removed Component Settings"? Then the `fixed_toolbar_container` line could be under "Removed Editor Core Settings" or something.

File: migration-from-4x.md (hunk 12 above)
Comment (blackcathikari) [reply migration-from-4x.md]: Also textcolor - see DOC-143

File: migration-from-4x.md (hunk 15 above)
Comment (blackcathikari) [reply migration-from-4x.md]: Done.

File: migration-from-4x.md (hunk 11 above)
Comment (blackcathikari) [reply migration-from-4x.md]: I spoke to @TheSpyder and we've decided to delete this. We'll add it back in if we can remember what it referred to.

File: migration-from-4x.md (hunk 1 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Done. Added a link to the Quickstart page.

File: migration-from-4x.md (hunk 2 above)
Comment (shikhanansi) [reply migration-from-4x.md]: I have provided information on what has been removed and changed in the chapter later.

File: migration-from-4x.md (hunk 3 above)
Comment (shikhanansi) [reply migration-from-4x.md]: @EkimChau confirmed that this information is accurate for TinyMCE 5.0.

File: migration-from-4x.md (hunk 4 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Done

File: migration-from-4x.md (hunk 5 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Added links and changed content wherever required.

File: migration-from-4x.md (hunk 6 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Added link to GitHub tickets.

File: migration-from-4x.md (hunk 7 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Reviewed and changed content.

File: migration-from-4x.md (hunk 8 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Link provided

File: migration-from-4x.md [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+
+### Custom Context Toolbars
+
+The Context Toolbar takes its buttons the registry of toolbar buttons added using `addButton`, `addToggleButton`, `addSplitButton` or `addMenuButton`. The method for creating custom context toolbars has also been moved from `editor.addContextToolbar()` to `editor.ui.registry.addContextToolbar()`.
+
+For more information on Context Toolbars, see [docs]({{site.baseurl}}/components/contexttoolbar).
+
+#### Changes between TinyMCE 4.x and TinyMCE 5.0:
+
+* Buttons go before and after the input in TinyMCE 4.x.
```
Comment (blackcathikari) [inline migration-from-4x.md]: We either need to fix this list or remove it.

File: migration-from-4x.md (hunk 9 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Reference link added.

File: migration-from-4x.md (hunk 14 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Fixed

File: migration-from-4x.md (hunk 11 above)
Comment (shikhanansi) [reply migration-from-4x.md]: I've added more context for explanation.

File: migration-from-4x.md (hunk 12 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Added colorpicker to the list. Will add textcolor after Beta.

File: migration-from-4x.md (hunk 13 above)
Comment (shikhanansi) [reply migration-from-4x.md]: Confirmed with @blackcathikari. It's correct

**Comment by shikhanansi** [comment]
Merging the branch into develop.
